sympy/solvers/solvers.py

```diff
@@ -1557,7 +1557,10 @@ def _expand(p):
                         try:
                             t = Dummy('t')
                             iv = _solve(u - t, symbol, **flags)
-                            soln = list(ordered({i.subs(t, s) for i in iv for s in soln}))
+                            new_soln = list(ordered({i.subs(t, s) for i in iv for s in soln}))
```

Comment:
I don't think this is right: if u is not a symbol, the solutions you have are not for the symbol they are the solutions for u. It is serendipitous that the solution for u in this case is also the solution for x.